File: backend/app/v1/schemas/skill.py
"""
Pydantic schemas for Skill-related data transfer.
"""

import uuid

from pydantic import BaseModel, ConfigDict, Field


class SkillBase(BaseModel):
    """
    Base schema for Skill data with shared attributes.
    """

    name: str
    description: str | None = None
    default_weightage: float | None = Field(default=10.0, description="Default weightage for this skill")


class SkillCreate(SkillBase):
    """
    Schema for creating a new Skill.
    """

    pass


class SkillUpdate(BaseModel):
    """
    Schema for updating an existing Skill.
    """

    name: str | None = None
    description: str | None = None
    default_weightage: float | None = Field(default=10.0, description="Default weightage for this skill")


class SkillRead(SkillBase):
    """
    Schema for reading Skill data, including database-generated fields.
    """

    id: uuid.UUID

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/skill.ts
import type { SkillCreateFormValues, SkillUpdateFormValues } from "@/schemas/skill";

/**
 * Base fields for a skill.
 */
export interface SkillBase {
  id: string;
  name: string;
  description?: string;
  default_weightage?: number;
}

/**
 * Payload for creating a new skill.
 */
export type SkillCreate = SkillCreateFormValues

/**
 * Payload for updating an existing skill.
 */
export type SkillUpdate = SkillUpdateFormValues

/**
 * Skill returned from read operations.
 */
export type SkillRead = SkillBase
